apps/snitch_api/test/snitch_api_web/controllers/line_item_controller_test.exs

```diff
@@ -0,0 +1,74 @@
+defmodule SnitchApiWeb.LineItemControllerTest do
+  use SnitchApiWeb.ConnCase, async: true
+  import Plug.Conn
+  alias Snitch.Repo
+  alias SnitchApi.Accounts
+  import Snitch.Factory
+  # alias Snitch.Data.Model.Order, as: OrderModel
+
+  setup %{conn: conn} do
+    conn =
+      conn
+      |> put_req_header("accept", "application/vnd.api+json")
+      |> put_req_header("content-type", "application/vnd.api+json")
+
+    user = build(:user_with_no_role)
+
+    role = build(:role, name: "user")
+
+    Repo.insert(role, on_conflict: :nothing)
+
+    {:ok, registered_user} = Accounts.create_user(user)
+
+    # create the token
+    {:ok, token, _claims} = SnitchApi.Guardian.encode_and_sign(registered_user)
+
+    # add authorization header to request
+    conn = conn |> put_req_header("authorization", "Bearer #{token}")
+
+    conn = assign(conn, :current_user, registered_user)
+
+    # pass the connection and the user to the test
+
+    {:ok, conn: conn}
+  end
+
+  describe "Line Items" do
+    test "Adding new item", %{conn: conn} do
+      user = conn.assigns[:current_user]
+      order = insert(:order, user_id: user.id)
+
+      variant =
+        insert(:variant, product: insert(:product), shipping_category: insert(:shipping_category))
+
+      line_item =
+        :line_item
+        |> build(unit_price: nil)
+        |> Map.take([:quantity, :unit_price])
```

Comment:
don't send `unit_price`, as it should be set from variant

Reply:
https://github.com/aviabird/snitch/blob/4f30d2579cb027b4052bde8f01b4954fce762808/apps/snitch_api/lib/snitch_api_web/controllers/line_item_controller.ex#L13
Exactly I did.

Reply:
same issue. I think the `insert()` functions are not inserting into the test DB. But, I am getting `order_id` and `variant_id`.  Could you confirm that they are inserting in your machine...

Reply:
ok

Reply:
I print out the `line_item` in line item controller.
Found out that unit_price is not setting using `update_unit_price`.
```map
%{
  "id" => nil,
  "order_id" => "173",
  "quantity" => "2",
  "type" => "line_item",
  "unit_price" => "",
  "variant_id" => "567"
}
```

Reply:
Try to debug that @blackode